mautic/mautic
# Adding support for URL shortener in email

* This feature was available only for links in SMS. However, there is legitimate interest for users to make this feature available also for links in E-Mails, especially when using a self-hosted URL shortener

* This feature is implemented such that if a link shortener URL is present, and the newly added configuration option to enable link shortening in e-Mails is set to "Yes", all tracking links and redirect links in e-mails are shortened using that link shortener URL

* This PR relates to the following discussions:

  * https://forum.mautic.org/t/url-shortener-how-and-where-does-it-work/16544
  * https://github.com/mautic/mautic/issues/2357
  * https://forum.mautic.org/t/allow-url-shortening-for-emails/17658
  * https://github.com/mautic/mautic/issues/39
  * https://github.com/mautic/mautic/issues/66
  * https://forum.mautic.org/t/how-to-track-clicks-in-sms/21267/5
  * https://github.com/mautic/mautic/pull/10280



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | []
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #2357 #66 #39 -->



#### Description:



#### Steps to test this PR:



1) In the Mautic Configuration, navigate to 'System Settings' -> 'Miscellaneous Settings'.
2) Add a valid 'URL Shortener' URL as specified in the tool tip, and enable URL shortening in e-mails (set to "Yes")
3) Create a 'Test' segment
4) Create a test contact
5) Add the test contact to the 'Test' segment
6) Create a new segment e-mail, and assign it to the 'Test' segment
7) In the e-mail body, add some links. These links are transformed into trackable redirect links, which we wish to **shorten**
8) Save and close the e-mail
9) "Send" the e-mail. This sends the e-mail to the contacts in the 'Test' segment, while generating and shortening tracking links
10) Open the e-mail in the test user's mailbox, and verify that the tracking links have been shortened
11) Disable short links for e-mail in the Mautic configuration, and repeat 4)-9)
12) Verify that links in sent E-Mails are not shortened
